src/PsrProducer.php

```diff
@@ -13,4 +13,55 @@
      * @throws InvalidMessageException     if an invalid message is specified
      */
     public function send(PsrDestination $destination, PsrMessage $message);
+
+    /**
+     * Sets the minimum length of time in milliseconds that must elapse after a message is sent before the provider may deliver the message to a consumer.
+     * deliveryDelay is set to zero by default.
+     *
+     * @param float|int $deliveryDelay - the delivery delay in milliseconds.
+     */
+    public function setDeliveryDelay($deliveryDelay);
+
+    /**
+     * Gets the minimum length of time in milliseconds that must elapse after a message is sent before the provider may deliver the message to a consumer.
+     *
+     * @return int|float the delivery delay in milliseconds.
+     */
+    public function getDeliveryDelay();
+
+    /**
+     * Specifies the priority of messages that are sent using this Producer
+     * The API defines ten levels of priority value, with 0 as the lowest priority and 9 as the highest.
+     * Clients should consider priorities 0-4 as gradations of normal priority and priorities 5-9 as gradations of expedited priority.
+     * Priority is set to 4 by default.
+     *
+     * @param int $priority
+     */
+    public function setPriority($priority);
```

Comment:
While in JMS spec `3.4.10   JMSPriority` it says

> JMS does not require that a provider strictly implement priority ordering of 
messages; however, it should do its best to deliver expedited messages ahead 
of normal messages.

is there a list of implementation that potentially may support this feature?